File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Source/Tva.Phasors/ConfigurationCellParsingState.vb
'*******************************************************************************************************
'  ConfigurationCellParsingState.vb - Configuration frame cell parsing state class
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: James R Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  01/14/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

' This class represents the protocol independent common implementation of a parsing state for a set of configuration related data settings that can be sent or received from a PMU.
<CLSCompliant(False)> _
Public Class ConfigurationCellParsingState

    Inherits ChannelCellParsingStateBase
    Implements IConfigurationCellParsingState

    Private m_createNewPhasorDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IPhasorDefinition)
    Private m_createNewFrequencyDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IFrequencyDefinition)
    Private m_createNewAnalogDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IAnalogDefinition)
    Private m_createNewDigitalDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IDigitalDefinition)

    Public Sub New( _
    ByVal createNewPhasorDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IPhasorDefinition), _
    ByVal createNewFrequencyDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IFrequencyDefinition), _
    ByVal createNewAnalogDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IAnalogDefinition), _
    ByVal createNewDigitalDefintionFunction As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IDigitalDefinition))

        m_createNewPhasorDefintionFunction = createNewPhasorDefintionFunction
        m_createNewFrequencyDefintionFunction = createNewFrequencyDefintionFunction
        m_createNewAnalogDefintionFunction = createNewAnalogDefintionFunction
        m_createNewDigitalDefintionFunction = createNewDigitalDefintionFunction

    End Sub

    Public Overrides ReadOnly Property InheritedType() As System.Type
        Get
            Return Me.GetType()
        End Get
    End Property

    Public ReadOnly Property CreateNewPhasorDefintionFunction() As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IPhasorDefinition) Implements IConfigurationCellParsingState.CreateNewPhasorDefintionFunction
        Get
            Return m_createNewPhasorDefintionFunction
        End Get
    End Property

    Public ReadOnly Property CreateNewFrequencyDefintionFunction() As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IFrequencyDefinition) Implements IConfigurationCellParsingState.CreateNewFrequencyDefintionFunction
        Get
            Return m_createNewFrequencyDefintionFunction
        End Get
    End Property

    Public ReadOnly Property CreateNewAnalogDefintionFunction() As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IAnalogDefinition) Implements IConfigurationCellParsingState.CreateNewAnalogDefintionFunction
        Get
            Return m_createNewAnalogDefintionFunction
        End Get
    End Property

    Public ReadOnly Property CreateNewDigitalDefintionFunction() As IConfigurationCellParsingState.CreateNewDefinitionFunctionSignature(Of IDigitalDefinition) Implements IConfigurationCellParsingState.CreateNewDigitalDefintionFunction
        Get
            Return m_createNewDigitalDefintionFunction
        End Get
    End Property

End Class

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/ConfigurationCellParsingState.cs
//*******************************************************************************************************
//  ConfigurationCellParsingState.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  01/14/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents the protocol independent common implementation of the parsing state used by any <see cref="IConfigurationCell"/>.
    /// </summary>
    public class ConfigurationCellParsingState : ChannelCellParsingStateBase, IConfigurationCellParsingState
    {
        #region [ Members ]

        // Fields
        private CreateNewDefinitionFunction<IPhasorDefinition> m_createNewPhasorDefinition;
        private CreateNewDefinitionFunction<IFrequencyDefinition> m_createNewFrequencyDefinition;
        private CreateNewDefinitionFunction<IAnalogDefinition> m_createNewAnalogDefinition;
        private CreateNewDefinitionFunction<IDigitalDefinition> m_createNewDigitalDefinition;

        #endregion

        #region [ Constructors ]

        /// <summary>
        /// Creates a new <see cref="ConfigurationCellParsingState"/> from specified parameters.
        /// </summary>
        /// <param name="createNewPhasorDefinition">Reference to delegate to create new <see cref="IPhasorDefinition"/> instances.</param>
        /// <param name="createNewFrequencyDefinition">Reference to delegate to create new <see cref="IFrequencyDefinition"/> instances.</param>
        /// <param name="createNewAnalogDefinition">Reference to delegate to create new <see cref="IAnalogDefinition"/> instances.</param>
        /// <param name="createNewDigitalDefinition">Reference to delegate to create new <see cref="IDigitalDefinition"/> instances.</param>
        public ConfigurationCellParsingState(CreateNewDefinitionFunction<IPhasorDefinition> createNewPhasorDefinition, CreateNewDefinitionFunction<IFrequencyDefinition> createNewFrequencyDefinition, CreateNewDefinitionFunction<IAnalogDefinition> createNewAnalogDefinition, CreateNewDefinitionFunction<IDigitalDefinition> createNewDigitalDefinition)
        {
            m_createNewPhasorDefinition = createNewPhasorDefinition;
            m_createNewFrequencyDefinition = createNewFrequencyDefinition;
            m_createNewAnalogDefinition = createNewAnalogDefinition;
            m_createNewDigitalDefinition = createNewDigitalDefinition;
        }

        #endregion

        #region [ Properties ]

        /// <summary>
        /// Gets reference to <see cref="CreateNewDefinitionFunction{T}"/> delegate used to create new <see cref="IPhasorDefinition"/> objects.
        /// </summary>
        public virtual CreateNewDefinitionFunction<IPhasorDefinition> CreateNewPhasorDefinition
        {
            get
            {
                return m_createNewPhasorDefinition;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewDefinitionFunction{T}"/> delegate used to create new <see cref="IFrequencyDefinition"/> objects.
        /// </summary>
        public virtual CreateNewDefinitionFunction<IFrequencyDefinition> CreateNewFrequencyDefinition
        {
            get
            {
                return m_createNewFrequencyDefinition;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewDefinitionFunction{T}"/> delegate used to create new <see cref="IAnalogDefinition"/> objects.
        /// </summary>
        public virtual CreateNewDefinitionFunction<IAnalogDefinition> CreateNewAnalogDefinition
        {
            get
            {
                return m_createNewAnalogDefinition;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewDefinitionFunction{T}"/> delegate used to create new <see cref="IDigitalDefinition"/> objects.
        /// </summary>
        public virtual CreateNewDefinitionFunction<IDigitalDefinition> CreateNewDigitalDefinition
        {
            get
            {
                return m_createNewDigitalDefinition;
            }
        }

        #endregion
    }
}